src/btree/bt_curnext.c

```diff
@@ -905,7 +905,12 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
  *     Move to the next record in the tree.
  */
 int
-__wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
+__wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating, bool evict_reposition)
 {
-    return (__wt_btcur_next_prefix(cbt, NULL, truncating));
+    WT_RET(__wt_btcur_next_prefix(cbt, NULL, truncating));
+
+    if (evict_reposition)
```

Comment:
Is there a reason why this check is "special" and not folded into the prefix version of `btcur_next` like other functionality? Phrased a different way - I believe `btcur_next` is pretty much just a shallow rapper around `btcur_next_prefix` now, to hide a parameter that's generally optional.

Adding functionality into `btcur_next` directly changes that relationship, which doesn't seem like the right choice to me.

Reply:
Agreed! Thanks, updated in [78bd0a0](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/78bd0a0ff3127cc5d437afb617856efe930fdd54)